extensions/ql-vscode/src/interface.ts

```diff
@@ -352,35 +352,39 @@ export class InterfaceManager extends DisposableObject {
       });
     }
 
-    const getParsedResultSets = async (): Promise<ParsedResultSets> => {
+    // Note that the resultSetSchemas will return offsets for the default (unsorted) page,
+    // which may not be correct. However, in this case, it doesn't matter since we only
+    // need the first offset, which will be the same no matter which sorting we use.
+    const resultSetSchemas = await this.getResultSetSchemas(results);
+    const resultSetNames = resultSetSchemas.map(schema => schema.name);
 
-      const resultSetSchemas = await this.getResultSetSchemas(results);
-      const resultSetNames = resultSetSchemas.map(schema => schema.name);
+    const selectedTable = getDefaultResultSetName(resultSetNames);
+    const schema = resultSetSchemas.find(
+      (resultSet) => resultSet.name == selectedTable
+    )!;
 
-      // This may not wind up being the page we actually show, if there are interpreted results,
-      // but speculatively send it anyway.
-      const selectedTable = getDefaultResultSetName(resultSetNames);
-      const schema = resultSetSchemas.find(
-        (resultSet) => resultSet.name == selectedTable
-      )!;
+    // Use sorted results path if it exists. This may happen if we are
+    // reloading the results view after it has been sorted in the past.
+    const resultsPath = results.sortedResultsInfo.get(selectedTable)?.resultsPath
```

Comment:
Would it be useful to have this as a helper function on `CompletedQuery`, so we can ensure we always use this logic when getting the results path?

Reply:
Yes. Done.